pkg/notifiers/sender.go

```diff
@@ -516,7 +516,7 @@ func NewClientFromConfig(cfg *v3.HTTPClientConfig, dialer dialer.Dialer) (*http.
 		Timeout: time.Second * 10,
 	}
 
-	if cfg != nil {
+	if cfg != nil && cfg.ProxyURL != "" {
```

Comment:
The utility function can be used here as well. We can define it in `pkg/notifiers` and use it in `pkg/controllers`